pkg/securitypolicy/securitypolicyenforcer_rego.go

```diff
@@ -21,50 +36,491 @@ func init() {
+		return err
+	}
+
+	if _, err := builder.WriteString(fmt.Sprintf("%s\"allow_elevated\": %v,\n", indent+Indent, container.AllowElevated)); err != nil {
+		return err
+	}
+
+	if _, err := builder.WriteString(fmt.Sprintf("%s\"working_dir\": \"%s\"\n", indent+Indent, container.WorkingDir)); err != nil {
+		return err
+	}
+
+	if _, err := builder.WriteString(fmt.Sprintf("%s}", indent)); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func addContainers(builder *strings.Builder, containers []*securityPolicyContainer) error {
+	if _, err := builder.WriteString("containers := [\n"); err != nil {
+		return err
+	}
+
+	for i, container := range containers {
+		if err := writeContainer(builder, container, Indent); err != nil {
+			return err
+		}
+
+		var end string
+		if i < len(containers)-1 {
+			end = ",\n"
+		} else {
+			end = "\n"
+		}
+
+		if _, err := builder.WriteString(end); err != nil {
+			return err
+		}
+	}
+
+	if _, err := builder.WriteString("]\n"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (p securityPolicyInternal) MarshalRego() (string, error) {
+	builder := new(strings.Builder)
+	if _, err := builder.WriteString(fmt.Sprintf("package policy\nallow_all := %v\n", p.AllowAll)); err != nil {
+		return "", err
+	}
+
+	if err := addContainers(builder, p.Containers); err != nil {
+		return "", err
+	}
+
+	return builder.String(), nil
+}
+
+func createRegoEnforcer(state SecurityPolicyState, defaultMounts []oci.Mount, privilegedMounts []oci.Mount) (SecurityPolicyEnforcer, error) {
+	policy, err := state.SecurityPolicy.toInternal()
+	if err != nil {
+		return nil, fmt.Errorf("error converting to internal format: %w", err)
+	}
+
+	regoPolicy, err := newRegoPolicyFromInternal(policy, defaultMounts, privilegedMounts)
+	if err != nil {
+		return nil, fmt.Errorf("error converting to Rego: %w", err)
+	}
+
+	regoPolicy.base64policy = state.EncodedSecurityPolicy.SecurityPolicy
+	return regoPolicy, nil
+}
+
+func newRegoPolicyFromInternal(securityPolicy *securityPolicyInternal, defaultMounts []oci.Mount, privilegedMounts []oci.Mount) (*RegoEnforcer, error) {
+	policy := new(RegoEnforcer)
+	policy.behavior = PolicyCode
+	if code, err := securityPolicy.MarshalRego(); err == nil {
+		policy.objects = code
+	} else {
+		return nil, fmt.Errorf("failed to convert json to rego: %w", err)
+	}
+
+	defaultMountData := make([]interface{}, 0, len(defaultMounts))
+	privilegedMountData := make([]interface{}, 0, len(privilegedMounts))
+	policy.data = map[string]interface{}{
+		"started":          []string{},
+		"devices":          map[string]string{},
+		"containers":       map[string]interface{}{},
+		"defaultMounts":    appendMountData(defaultMountData, defaultMounts),
+		"privilegedMounts": appendMountData(privilegedMountData, privilegedMounts),
+		"sandboxPrefix":    guestpath.SandboxMountPrefix,
+		"hugePagesPrefix":  guestpath.HugePagesMountPrefix,
+	}
+	policy.mutex = new(sync.Mutex)
```

Comment:
The default value of a `sync.Mutex` is ready for use, so unless we specifically need it to be a pointer for some reason, you can just change the type to `mutex sync.Mutex`, and don't need this line.

Reply:
I think this was done to mirror the usage of a pointer that I did in the standard security policy enforcer. I've changed here.